### PR TITLE
Make sure all new docker images are also tagged with commit

### PR DESCRIPTION
**Pull request by jakubgs**
Since we want to pull docker images for the cluster based on commit hardcoded into Ansible configuration(for now) we need to make sure that images pushed to Docker Hub are always tagges with the commit as well.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -129,15 +131,23 @@ statusgo-library: ##@cross-compile Build status-go as static library for current
...
-	docker build --file _assets/build/Dockerfile-bootnode . -t $(BOOTNODE_IMAGE_NAME):latest
-
-docker-image-tag: ##@docker Tag DOCKER_IMAGE_NAME:latest with a tag following pattern $GIT_SHA[:8]-$BUILD_TAGS
-	@echo "Tagging docker image..."
-	docker tag $(DOCKER_IMAGE_NAME):latest $(DOCKER_IMAGE_NAME):$(shell BUILD_TAGS="$(BUILD_TAGS)" ./_assets/ci/get-docker-image-tag.sh)
+	docker build --file _assets/build/Dockerfile-bootnode . \
+		-t $(BOOTNODE_IMAGE_NAME):$(DOCKER_IMAGE_CUSTOM_TAG) \
+		-t $(BOOTNODE_IMAGE_NAME):latest
+
+push-docker-images: docker-image bootnode-image
```
Comment (adambabik) [inline Makefile:149]: This method may bring a lot of confusion if run by someone who forgets to update the branch or runs it from a different branch than `develop`. I would recommend pushing images with `$(DOCKER_IMAGE_CUSTOM_TAG)` tags and have a dedicated command to push `latest` tag which verifies it's not `develop` and it's up-to-date.

Comment (jakubgs) [reply Makefile:149]: That's fair, will change it.

Comment (jakubgs) [reply Makefile:149]: Although, what do you mean when you say "verifies it's not `develop`"? You mean `latest` tag should never be updated form the `develop` branch?

**Comment by divan** [comment]
So this removes `docker-image-tag` target in favour of `push-docker-images`, because tagging without pushing anyway doesn't make much sense, right?

**Review by divan — APPROVED** [review]
LGTM

**Comment by jakubgs** [comment]
Well, to be exact it tags both with `latest` and the commit at build stage, but yeah, the point is to not push images without commit tag.

File: Makefile (hunk 1 above)
Comment (adambabik) [reply Makefile:149]: Oh sorry. I meant "verifies it IS `develop`". So `latest` tag should actually be only created from the main branch which in our case is `develop`.

**Comment by adambabik** [comment]
> Well, to be exact it tags both with latest and the commit at build stage, but yeah, the point is to not push images without commit tag.

I think the point is to push images with a commit tag and pushing `latest` should be manual for now or a separate target with the checks (1) if it's `develop` branch and (2) it's up-to-date with the origin.

File: Makefile (hunk 1 above)
Comment (jakubgs) [reply Makefile:149]: Added a check.

**Comment by jakubgs** [comment]
I've added a check to that task. If you want I can add even some kind of confirmation step.

**Review by adambabik — APPROVED** [review]
LGTM!

The only thing we can add to make it even safer is to check if there are no new commits with `git status -s -u no` when pushing the `latest` tags.

**Comment by jakubgs** [comment]
@adambabik don't you mean there are no un-commited files?

**Comment by adambabik** [comment]
No, I mean that there are no new commits in the upstream. It could happen that someone pushed a new image just before you were about to do it and your `develop` is now outdated.

**Comment by jakubgs** [comment]
How about now @adambabik?

File: Makefile [hunk 2]
```diff
@@ -28,6 +28,9 @@ endif
 CGO_CFLAGS=-I/$(JAVA_HOME)/include -I/$(JAVA_HOME)/include/darwin
 GOBIN=$(dir $(realpath $(firstword $(MAKEFILE_LIST))))build/bin
 GIT_COMMIT := $(shell git rev-parse --short HEAD)
+GIT_BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+GIT_LOCAL  := $(git rev-parse @)
```
Comment (adambabik) [inline Makefile]: `shell` may be required before `git rev-parse @` and `git rev-parse origin`

File: Makefile [hunk 3]
```diff
@@ -129,15 +134,33 @@ statusgo-library: ##@cross-compile Build status-go as static library for current
...
+	docker build --file _assets/build/Dockerfile-bootnode . \
+		-t $(BOOTNODE_IMAGE_NAME):$(DOCKER_IMAGE_CUSTOM_TAG) \
+		-t $(BOOTNODE_IMAGE_NAME):latest
+
+push-docker-images: docker-image bootnode-image
+	docker push $(BOOTNODE_IMAGE_NAME):$(DOCKER_IMAGE_CUSTOM_TAG)
+	docker push $(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_CUSTOM_TAG)
+
+push-docker-images-latest: docker-image bootnode-image
+ifneq ("$(GIT_BRANCH)", "develop")
```
Comment (adambabik) [inline Makefile:154]: I would call `git fetch origin` first to make sure we're up-to-date with the upstream.

Comment (jakubgs) [reply Makefile:154]: Good point!

File: Makefile (hunk 2 above)
Comment (jakubgs) [reply Makefile]: yeah

**Pull request merged**